### PR TITLE
Move welcome function to python3-http

### PR DESCRIPTION
**Pull request by kturcios**
Signed-off-by: Kevin Turcios <kevin_turcios@outlook.com>

### CODE REVIEW

File: join-welcome/handler.py [hunk 1]
```diff
@@ -6,71 +6,79 @@
...
+    # TODO implement
+    log_event(event.body)
 
     if os.getenv("log_env", "0") == "1":
         log_env()
 
     r = None
     try:
-        r = json.loads(req)
+        r  = json.loads(event.body)
```
Comment (alexellis) [inline join-welcome/handler.py]: extra space?

File: join-welcome/handler.py [hunk 2]
```diff
@@ -6,71 +6,79 @@
...
-
-def handle(req):
-    """handle a request to the function
-    Args:
-        req (str): request body
-    """
-
-    log_event(req)
+def handle(event, context):
+    # TODO implement
```
Comment (alexellis) [inline join-welcome/handler.py]: Don't need

File: join-welcome/handler.py [hunk 3]
```diff
@@ -6,71 +6,79 @@
...
     if os.getenv("log_env", "0") == "1":
         log_env()
 
     r = None
     try:
-        r = json.loads(req)
+        r  = json.loads(event.body)
     except ValueError:
         sys.stderr.write("Error parsing request, invalid JSON")
-        return "Error parsing request, invalid JSON"
```
Comment (alexellis) [inline join-welcome/handler.py:27]: Give it a status code too.. 400?

File: join-welcome/handler.py [hunk 4]
```diff
@@ -6,71 +6,79 @@
...
+    target_channel = os.getenv("target_channel")
+    digest = ""
+    if "Http_X_Slack_Signature" in event.headers:
+        digest = event.headers["Http_X_Slack_Signature"]
 
-        target_channel = os.getenv("target_channel")
-        digest = os.getenv("Http_X_Slack_Signature", "")
+    # Takes format of: "Http_X_Slack_Signature v0=hash"
+    slack_request_timestamp = ""
+    if "Http_X_Slack_Request_Timestamp" in event.headers:
```
Comment (alexellis) [inline join-welcome/handler.py]: `Http_X_Slack_Request_Timestamp` -> `X-Slack-Request-Timestamp`

Comment (alexellis) [reply join-welcome/handler.py]: Since we read this N times, let's make a const / string for it

File: join-welcome/handler.py (hunk 3 above)
Comment (kturcios) [reply join-welcome/handler.py:27]: I thought I did return a 400? This is for the removed code

Comment (alexellis) [reply join-welcome/handler.py:27]: That looks like returning a string?

File: join-welcome/handler.py (hunk 4 above)
Comment (kturcios) [reply join-welcome/handler.py]: Shall I do it for both `Http_X_Slack_Signature` and `X_Slack_Request_Timestamp`?

File: join-welcome/handler.py (hunk 3 above)
Comment (kturcios) [reply join-welcome/handler.py:27]: This is what is being returned no?
```python
try:
        r = json.loads(event.body)
except ValueError:
        sys.stderr.write("Error parsing request, invalid JSON")
        return {
            "statusCode": 400,
            "body": "Error parsing request, invalid JSON"
        }
```

File: join-welcome/handler.py (hunk 4 above)
Comment (alexellis) [reply join-welcome/handler.py]: All of them need changing to the format as they appear in the HTTP header in Flask.

`Http_X_Thing_Here` -> `X-Thing-Here`

File: join-welcome/handler.py (hunk 3 above)
Comment (alexellis) [reply join-welcome/handler.py:27]: My bad. I misread it.

File: join-welcome/handler.py (hunk 4 above)
Comment (kturcios) [reply join-welcome/handler.py]: Updated

File: join-welcome/handler.py [hunk 5]
```diff
@@ -6,71 +6,81 @@
...
-    Args:
-        req (str): request body
-    """
-
-    log_event(req)
+def handle(event, context):
+    SLACK_SIG_HEADER = "X-Slack-Signature"
+    SLACK_TIMESTAMP_HEADER = "X-Slack-Request-Timestamp"
+    
+    log_event(event.body)
```
Comment (alexellis) [inline join-welcome/handler.py:13]: Is event.body JSON or the raw value? It will affect the hmac results

Comment (kturcios) [reply join-welcome/handler.py:13]: Nope its a single bytestring.
http://flask.pocoo.org/docs/1.0/api/#flask.Request.get_data

**Comment by alexellis** [comment]
Let's try this

**Pull request merged**